hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/streaming/HoodieSourceOffset.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hudi.streaming
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+import org.apache.spark.sql.execution.streaming.{Offset, SerializedOffset}
+import org.apache.spark.sql.sources.v2.reader.streaming.{Offset => OffsetV2}
```

Comment:
Thanks for your suggestion.I have test with -Dspark3. The `HoodieSourceOffset`  is not work for spark3.
 I have a puzzle that should we leave it in `hudi-spark` or `hudi-spark2`? Currently most of the Spark code stay in the `hudi-spark`. If  moving this to `hudi-spark2` now, many other relate code should also move. So I think we can create a JIRA to track this currently. In the long term, I will provide an implement for the spark3.

Reply:
If it's only compatible with Spark 2, we should leave it in ```hudi-spark2```. If it's compatible with both Spark2 and Spark3, we should leave it in ```hudi-spark```. 
My understand is at least you need to make this change be able to compile with ```mvn packge -DskipTests -Dspark3```. Otherwise Hudi cannot compile with Spark 3 any more.

Reply:
Thanks @zhedoubushishi for your advice. I will do fix this problem later.

Reply:
done for this!